src/client.cpp

```diff
@@ -119,6 +119,7 @@ CClient::CClient ( const quint16  iPortNumber,
     QObject::connect ( &Channel, &CChannel::ReqChanInfo, this, &CClient::OnReqChanInfo );
 
     QObject::connect ( &Channel, &CChannel::ConClientListMesReceived, this, &CClient::ConClientListMesReceived );
+    QObject::connect ( &Channel, &CChannel::ConClientListMesReceived, this, &CClient::OnConClientListMesReceived );
```

Comment:
I'm a bit unsure what is better here:
- Use a single method to handle both cases (would require to re-emit the signal though, I think?)
- or use two `connect()` calls for those unrelated jobs.

I chose the latter as the jobs are unrelated (besides sharing the same trigger).

Reply:
Two handlers is better.

Reply:
This, basically, is the fundamental flaw in the current design.

The Client is simply reemitting the signal on line 123 without "knowing" about it.  I think it would be a better approach for the Client to _handle_ all signals, hold state for what they "mean" and have GUI/JSONRPC/etc send state updates as signals.  The client could them emit _specific_ state change messages (i.e. in this case, which channel changed, one emit per channel, along with the new state).

However, for now, I'd be quite happy with either approach - either the new handler emitting `CClient::ConClientListMesReceived` when complete or the one taken here.